cal-itp/littlepay
# Feat: basic configuration setup

## What this PR does

* Adds a simple YAML-based config file, wrapper class, and commands to interact with the config
* Replaces the `info` command with `config` as the default
* Adds some basic usage docs in the README
* Adds [`pytest-watcher`](https://github.com/olzhasar/pytest-watcher) to the devcontainer to keep tests running in the background `<--` LMK what we think of this? Kind of cool, kind of annoying?


## Testing this PR

1. Checkout this branch
2. Rebuild and Reopen the devcontainer
3. Notice a terminal is opened, awaiting file changes to run tests -- open any Python file and click `Save` to trigger
4. Run tests from VS Code test runner
5. Call the CLI inside the devcontainer:
 
  _Print help info_
  ```console
  littlepay -h
  ```
  _Initialize a config file_
  ```console
  littlepay config
  ```
  _Switch to the prod environment_
  ```console
  littlepay switch env prod
  ```
